projects/clr/rocclr/platform/command.hpp

```diff
@@ -1383,29 +1383,35 @@ class AccumulateCommand : public Command {
  private:
   //! Kernel names and timestamps list for activity profiling
   std::vector<std::string> kernelNames_;
+  const std::vector<std::string>* kernelNamesRef_;
   std::vector<std::pair<uint64_t, uint64_t>> tsList_;
 
  public:
   //! Create a new Marker
   AccumulateCommand(HostQueue& queue, const EventWaitList& eventWaitList = nullWaitList,
                     const Event* waitingEvent = nullptr)
-      : Command(queue, CL_COMMAND_TASK, eventWaitList, 0, waitingEvent) {}
+      : Command(queue, CL_COMMAND_TASK, eventWaitList, 0, waitingEvent), kernelNamesRef_(nullptr) {}
 
-  //! Add kernel name to the list if available
-  void addKernelName(const std::string& kernelName) { kernelNames_.push_back(kernelName); }
+  //! Set external kernel names reference
+  void setKernelNamesRef(const std::vector<std::string>* kernelNames) {
+    kernelNamesRef_ = kernelNames;
+  }
 
-  //! Add multiple kernel names in bulk
-  void addKernelNames(const std::vector<std::string>& kernelNames) {
-    kernelNames_.insert(kernelNames_.end(), kernelNames.begin(), kernelNames.end());
+  //! Move kernel names into internal storage and set reference to it
+  void moveKernelNames(std::vector<std::string>&& kernelNames) {
```

Comment:
I dont understand why you need move and setKernelNames, this is messy, to be clean one only needs references to kernel names , which are in codeobject. that way we dont copy  copy any string around.